### PR TITLE
Remove trash button in post/page inspector

### PR DESCRIPTION
**Pull request by jameskoster**
Part of https://github.com/WordPress/gutenberg/issues/59689.

## What?
Remove the 'Move to trash' button beneath post meta.

## Why?
1. It's overly prominent.
2. Inconsistent with other trashing flows (e.g. templates and patterns).
3. Duplicative with the new actions menu.

| Before | After |
| --- | --- |
| Screenshot 2024-05-20 at 14 13 34 |  Screenshot 2024-05-20 at 14 13 01 |

## Testing Instructions
1. Open a post or page.
2. Open the Inspector and select the Post/Page tab.
3. Check the 'Move to trash' button is no longer visible.
4. Check the 'Move to trash' action in the ellipsis menu works as expected.
5. Check other post types (pattern, template, navigation menu).

### CODE REVIEW

**Comment by richtabor** [comment]
Should the trash button still have the destructive treatment?

File: packages/editor/src/components/index.js [hunk 1]
```diff
@@ -81,8 +81,6 @@ export { default as PostTaxonomiesPanel } from './post-taxonomies/panel';
 export { default as PostTextEditor } from './post-text-editor';
 export { default as PostTitle } from './post-title';
 export { default as PostTitleRaw } from './post-title/post-title-raw';
-export { default as PostTrash } from './post-trash';
-export { default as PostTrashCheck } from './post-trash/check';
```
Comment (youknowriad) [inline packages/editor/src/components/index.js]: These components should be kept here as they are public APIs.

**Comment by jameskoster** [comment]
> Should the trash button still have the destructive treatment?

Probably not in this case because it isn't truly destructive (the post can be retrieved from trash). Also; no other menu items of this type (deleting template / patterns) are styled as destructive. Edit: I lied, the one in the dark details panel is destructive, but that one's being removed soon 😅

FWIW I've seen @jasmussen advocate _not_ using 'destructive' colorisation because in some cultures red != destructive. It might be interesting to add some smarts so that color is applied to destructive actions based on `lang`, but obviously that's separate.

**Review by ntsekouras — APPROVED** [review]
LGTM, thanks!

**Comment by Mamaduka** [comment]
It's unclear if we'll be able to ship "Post Actions" with the next major WP release; the last Gutenberg version RC that goes in the core is scheduled for next week.

I just wanted to note that we might have to restore this button.

**Comment by jasmussen** [comment]
> Should the trash button still have the destructive treatment?

> FWIW I've seen @jasmussen advocate not using 'destructive' colorisation because in some cultures red != destructive. It might be interesting to add some smarts so that color is applied to destructive actions based on lang, but obviously that's separate.

I do think we're using the "destructive" treatment without good semantics for when and what. Consider this screenshot showing a published post:

published post inspector

Here, "move to trash" is denoted destructive, but "switch to draft" — arguably much more destructive on a production site — is not. Does that mean they should both be red? Or just Switch to draft? 

The color change is also not useful for people with color blindness, nor is its meaning universally destructive across the world. Perhaps most importantly, the color alone is not going to be the best safeguard against performing truly destructive actions (such as empty trash or, as noted, unpublish post). In those cases, an "are you sure" modal that informs you that this is destructive, would almost certainly be better.

All that is to say that at the moment we are applying the color without a great heuristic, it's not clear it's that helpful to those who perceive color and recognize it as destructive, and in some cases the color is likely to simply be confusing: why is it red now? So more than anything, the color feels decorative rather than useful. Not a hill I'll die on, not something I'd want to block. But I think we can do better.

In this particular case, simply moving it to the ellipsis menu strikes a good balance between:

* Providing a safeguard in the shape of an extra click. 
* Reducing the prominence, so a mildly destructive action is not the most emphasized action in the panel.

**Comment by ntsekouras** [comment]
>It's unclear if we'll be able to ship "Post Actions" with the next major WP release; the last Gutenberg version RC that goes in the core is scheduled for next week.

@Mamaduka why is that? Have I missed some discussion or issue somewhere?

**Comment by Mamaduka** [comment]
@ntsekouras, proper permission checks are needed for one. I've already mentioned this a few times and even started exploring a solution, but it's nowhere finished - #61644.

**Comment by youknowriad** [comment]
@Mamaduka There's a simpler solution for now, just add the permissions checks in the `PostActions` component and omit the actions that are not allowed.

**Comment by Mamaduka** [comment]
@youknowriad sounds good. I am happy to help with testing/reviews.

**Pull request merged**